internal/ethapi/api.go

```diff
@@ -1392,6 +1396,10 @@ type SendTxArgs struct {
 
 // setDefaults is a helper function that fills in default values for unspecified tx fields.
 func (args *SendTxArgs) setDefaults(ctx context.Context, b Backend) error {
+	// Reject if Celo-only fields set when EthCompatible is true
+	if args.EthCompatible && !(args.FeeCurrency == nil && args.GatewayFeeRecipient == nil && (args.GatewayFee == nil || args.GatewayFee.ToInt().Sign() == 0)) {
```

Comment:
again we have duplicated "validation logic" here. I understand part of the problem is that `msg` is an interface and there are several implementers? Not sure.

Reply:
Yeah, there are several different types (and they don't even all have the same underlying fields, e.g. `SendTxArgs.GatewayFee` is `hexutil.Big` which is a wrapper around `big.Int`.  But I like the idea of moving it out to a method.  It will remove at least once case of duplication.  And, even where it doesn't, it would still be nicer.